Templates/templates/PHP/Model/EntityType.php.tt

```diff
@@ -114,7 +114,7 @@ foreach(var property in entity.Properties.Where(prop => prop.Type.GetTypeString(
     *
     * @param <#=propertyTypeReference#> $val The <#=propertyName#>
     *
-    * @return <#=entityCheckedCase#>
+    * @return <#=entityName.ToCheckedCase()#>
```

Comment:
We could consider `@return $this` here & on the other setters returning $this. More explicit that we're returning the instance of the current object (https://docs.phpdoc.org/3.0/guide/guides/types.html).

Reply:
Any reasons for the choice of `$this` over just the class name?

Reply:
More specificity. Shows that the method not only returns an instance of the class but the same instance the method was called on
Helpful also if somebody overrides a class. `return $this` allows the IDE to suggest/autocomplete the inherited setters.